clang/test/CodeGen/X86/avx10_2_512minmax-builtins.c

```diff
@@ -0,0 +1,244 @@
+  // CHECK-LABEL: @test_mm512_mask_minmax_ph(
+  // CHECK: call <32 x half> @llvm.x86.avx10.mask.vminmaxph.round(
+  return _mm512_mask_minmax_ph(__A, __B, __C, __D, 127);
+}
+
+__m512h test_mm512_maskz_minmax_ph(__mmask32 __A, __m512h __B, __m512h __C) {
+  // CHECK-LABEL: @test_mm512_maskz_minmax_ph(
+  // CHECK: call <32 x half> @llvm.x86.avx10.mask.vminmaxph.round(
+  return _mm512_maskz_minmax_ph(__A, __B, __C, 127);
+}
+
+__m512h test_mm512_minmax_round_ph(__m512h __A, __m512h __B) {
+  // CHECK-LABEL: @test_mm512_minmax_round_ph(
+  // CHECK: call <32 x half> @llvm.x86.avx10.mask.vminmaxph.round(
+  return _mm512_minmax_round_ph(__A, __B, 127, _MM_FROUND_NO_EXC);
+}
+
+__m512h test_mm512_mask_minmax_round_ph(__m512h __A, __mmask32 __B, __m512h __C, __m512h __D) {
+  // CHECK-LABEL: @test_mm512_mask_minmax_round_ph(
+  // CHECK: call <32 x half> @llvm.x86.avx10.mask.vminmaxph.round(
+  return _mm512_mask_minmax_round_ph(__A, __B, __C, __D, 127, _MM_FROUND_NO_EXC);
+}
+
+__m512h test_mm512_maskz_minmax_round_ph(__mmask32 __A, __m512h __B, __m512h __C) {
+  // CHECK-LABEL: @test_mm512_maskz_minmax_round_ph(
+  // CHECK: call <32 x half> @llvm.x86.avx10.mask.vminmaxph.round(
+  return _mm512_maskz_minmax_round_ph(__A, __B, __C, 127, _MM_FROUND_NO_EXC);
+}
+
+__m512 test_mm512_minmax_ps(__m512 __A, __m512 __B) {
+  // CHECK-LABEL: @test_mm512_minmax_ps(
+  // CHECK: call <16 x float> @llvm.x86.avx10.mask.vminmaxps.round(
+  return _mm512_minmax_ps(__A, __B, 127);
+}
+
+__m512 test_mm512_mask_minmax_ps(__m512 __A, __mmask16 __B, __m512 __C, __m512 __D) {
+  // CHECK-LABEL: @test_mm512_mask_minmax_ps(
+  // CHECK: call <16 x float> @llvm.x86.avx10.mask.vminmaxps.round(
+  return _mm512_mask_minmax_ps(__A, __B, __C, __D, 127);
+}
+
+__m512 test_mm512_maskz_minmax_ps(__mmask16 __A, __m512 __B, __m512 __C) {
+  // CHECK-LABEL: @test_mm512_maskz_minmax_ps(
+  // CHECK: call <16 x float> @llvm.x86.avx10.mask.vminmaxps.round(
+  return _mm512_maskz_minmax_ps(__A, __B, __C, 127);
+}
+
+__m512 test_mm512_minmax_round_ps(__m512 __A, __m512 __B) {
+  // CHECK-LABEL: @test_mm512_minmax_round_ps(
+  // CHECK: call <16 x float> @llvm.x86.avx10.mask.vminmaxps.round(
+  return _mm512_minmax_round_ps(__A, __B, 127, _MM_FROUND_NO_EXC);
+}
+
+__m512 test_mm512_mask_minmax_round_ps(__m512 __A, __mmask16 __B, __m512 __C, __m512 __D) {
+  // CHECK-LABEL: @test_mm512_mask_minmax_round_ps(
+  // CHECK: call <16 x float> @llvm.x86.avx10.mask.vminmaxps.round(
+  return _mm512_mask_minmax_round_ps(__A, __B, __C, __D, 127, _MM_FROUND_NO_EXC);
+}
+
+__m512 test_mm512_maskz_minmax_round_ps(__mmask16 __A, __m512 __B, __m512 __C) {
+  // CHECK-LABEL: @test_mm512_maskz_minmax_round_ps(
+  // CHECK: call <16 x float> @llvm.x86.avx10.mask.vminmaxps.round(
+  return _mm512_maskz_minmax_round_ps(__A, __B, __C, 127, _MM_FROUND_NO_EXC);
+}
+
+__m128d test_mm_minmax_sd(__m128d __A, __m128d __B) {
```

Comment:
Move scalar tests to avx10_2minmax-builtins.c

Reply:
[adfe6cd](https://github.com/llvm/llvm-project/pull/101598/commits/adfe6cd518f7f75e8dbbdfcdc6ddadae1a27fc4a)